nomad/structs/job.go

```diff
@@ -16,6 +24,53 @@ const (
 	JobServiceRegistrationsRPCMethod = "Job.GetServiceRegistrations"
 )
 
+// JobBatchDeregisterRequest is used to batch deregister jobs and upsert
+// evaluations.
+type JobBatchDeregisterRequest struct {
+
+	// Jobs is the set of jobs to deregister.
+	Jobs map[NamespacedID]*JobDeregisterOptions
+
+	// Evals is the set of evaluations to create.
+	//
+	// Deprecated: the job batch deregister endpoint no longer generates an
+	// eval per job.
+	Evals []*Evaluation
```

Comment:
I think we can safely remove this. If we send the new request to an older server and it's missing the field it'll be treated as empty on the other side. And if we send the older request to a new server, I'm pretty sure that new field gets ignored? Normally that'd be unsafe but we just end up dropping the behavior of writing new Evaluations, so that should be ok.

Worth testing across an upgrade though.

Reply:
Updated within: https://github.com/hashicorp/nomad/pull/20510/commits/039bf6aca7f8f83e5f783aa7760f33446e7cc1cd

Tested upgrades in AWS without issue.